setup.py

```diff
@@ -6,30 +6,37 @@
 from __future__ import absolute_import, print_function
 
 import os
-import re
 from distutils.sysconfig import get_config_var, get_python_inc
 
 import versioneer
 from Cython.Distutils.build_ext import new_build_ext as build_ext
 from setuptools import setup
 from setuptools.extension import Extension
 
-include_dirs = [os.path.dirname(get_python_inc())]
-library_dirs = [get_config_var("LIBDIR")]
-libraries = ["ucp", "uct", "ucm", "ucs"]
-extra_compile_args = ["-std=c99", "-Werror"]
+# Patch versioneer version for wheel builds.
+if "RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE" in os.environ:
+    orig_get_versions = versioneer.get_versions
+
+    version_override = os.environ["RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE"]
+    if not version_override:
+        raise RuntimeError(
+            "An empty RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE is not supported. "
+            "Either remove this variable from your environment or specify a "
+            "valid version override."
+        )
 
+    def get_versions():
+        data = orig_get_versions()
+        data["version"] = version_override
+        return data
 
-def get_ucp_version():
-    with open(include_dirs[0] + "/ucp/api/ucp_version.h") as f:
-        ftext = f.read()
-        major = re.findall("^#define.*UCP_API_MAJOR.*", ftext, re.MULTILINE)
-        minor = re.findall("^#define.*UCP_API_MINOR.*", ftext, re.MULTILINE)
+    versioneer.get_versions = get_versions
```

Comment:
I'll defer to @wence- here, but the answer here is related to his answer to your other query about why not directly use the version: this hack ensures that we get the right version encoded in `_version.py` so that `ucp.__version__` is correct when users import and use it.

Reply:
Yes, it's all part of the same dance.

The other approach would be to rewrite `_version.py` in the wheel-building process to fill in the version by hand. But this is kind of what versioneer does any, so this monkeypatch is simpler.